build.rs

```diff
@@ -619,7 +619,10 @@ fn cc(file: &Path, ext: &str, target: &Target, include_dir: &Path, out_file: &Pa
                                    // run-time checking: (s)tack frame, (u)ninitialized variables
             let _ = c.flag("/RTCsu");
         } else {
-            let _ = c.flag("/Ox"); // Enable full optimization.
```

Comment:
Do you agree that cc-rs does the right thing automatically, so that we can just remove both this line and the analogous "/Od" line above? I would prefer to do that, if cc-rs is automatically doing the right thing. (In general, I want to remove as much logic from build.rs that duplicates that cc-rs does as possible; see recent PRs that change build.rs to this effect.)

Reply:
I haven't previously used cc-rs so I assumed manually specifying optimization flags was somehow necessary 😅  
Well, cc-rs can determine the correct flag for each compiler: https://github.com/rust-lang/cc-rs/blob/2447a2ba5f455c00b1563193e125b60eebbd8ebe/src/lib.rs#L1748-L1749

I quickly checked the docs and there's [this function](https://docs.rs/cc/latest/cc/struct.Build.html#method.opt_level) that lets you manually specify the optimization level, but it also says that the `OPT_LEVEL` environment variable is used to determine the correct value otherwise (which is what is done manually here).

About `"/Od"`, it is the default so no need to specify it: https://learn.microsoft.com/en-us/cpp/build/reference/od-disable-debug?view=msvc-170

It's not clear to me whether `"/RTCsu"` is on by default:
https://learn.microsoft.com/en-us/cpp/build/reference/rtc-run-time-error-checks?view=msvc-170

The Visual Studio IDE does enable it for debug builds but that might not be true when using the compiler directly. We might need to read a little more, or just keep it to be sure.

tl;dr  
Yes, I totally agree, no optimization flags should be manually set unless there's a good reason for doing it. The `"/RTCsu"` situation is not 100% clear to me

Reply:
I took another look, but I couldn't conclusively find anything indicating that `/RTCsu` is enabled by default. Therefore I left that part but removed the rest. If you think you don't need `/RTCsu`, then the whole if statement can be removed, otherwise I would say the change is ready to be merged

Reply:
@andoalon I would be happy to have the /RTCsu removed.